pkg/ignition/builder.go

```diff
@@ -98,6 +97,11 @@ func (b *ignitionBuilder) Generate() ([]byte, error) {
 		})
 	}
 
+	config.Storage.Files = append(config.Storage.Files, ignitionFileEmbed(
```

Comment:
Let's please only create this file if there are settings to apply. We're working on integration with the assisted service, I'd like to be able to minimize the impact on them (e.g. if they have their own default environment).

Reply:
If there is a conflict, better that we have to deal with it all the time IMHO. Test coverage for proxy variables is notoriously poor.

Reply:
I'm not sure I get it... why do we need to create this file when no proxy settings are provided?

Reply:
We don't, but if creating it causes a bug then we don't want to find out only when it is used in the field, which is how we've found bugs with the proxy config in the past.